etc/init.d/zfs-import.in

```diff
@@ -308,8 +308,7 @@ do_start()
 
 # ----------------------------------------------------
 
-if [ ! -e /sbin/openrc-run ]
-then
+if ! (echo @SHELL@ | grep openrc 1>/dev/null 2>/dev/null); then
```

Comment:
Can we instead check for `/sbin/openrc-run` and what's currently used on Debian?  Additionally, these should be run time checks if at all possible so the specific configuration of the build system won't result in different scripts.

Reply:
Debian always use `/bin/sh` as the shell for init scripts, even if `/sbin/openrc-run` has been installed. At least `[ ! -e /sbin/openrc-run ]` is not enough. We discussed about this at https://github.com/zfsonlinux/zfs/issues/8204#issuecomment-450500048 but I don't know if there is a better solution than this PR.

Reply:
Since we can't detect this at runtime, can we instead address it on the downstream Debian ZFS packaging?

Reply:
This patch has been used by the Debian package long time ago.

Reply:
Great, then it sounds like there's no pressing need to include it upstream where we might encounter unexpected side effects on other platforms.

Reply:
Keeping the patch downstream sounds fine to me. Shall I close the PR?

Reply:
Yes, I think that would be best for now.  Thanks!